src/plugins/ui_core_views/cell_evaluation/evaluation_plugin.ts

```diff
@@ -309,6 +315,23 @@ export class EvaluationPlugin extends UIPlugin {
       .every((cell) => cell.type === CellValueType.empty);
   }
 
+  /**
+   * Fitler some values according to the hidden columns or row in the corresponding zone
+   * @param sheetId - The UID of the sheet to take into account
+   * @param zone - The zone to tke into account when filtering
+   * @param values - The original values to filter
+   * @returns the values filtered (ie we keep only the not hidden values)
+   */
+  private filterHiddenColsRows<T>(sheetId: UID, zone: Zone, values: T[]): T[] {
+    return positions(zone)
```

Comment:
I amended your commit with the missing UNDO/REDO commands and added a suggestion commit  because I felt like the map> filter > map was a bit hard to follow ^^'  Let me know what you think :)